### PR TITLE
CB-9237 stack can stuck, somehow status is deleted but terminated timestamp is null

### PR DESCRIPTION
**Pull request by sodre90**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TerminationService.java [hunk 1]
```diff
@@ -114,12 +114,18 @@ private void deleteOnlyIfForced(Stack stack, boolean force, String terminatedNam
...
-        if (stack.getType().equals(StackType.WORKLOAD)) {
-            grpcUmsClient.notifyResourceDeleted(stack.getResourceCrn(), MDCUtils.getRequestId());
+        try {
+            transactionService.required(() -> {
+                stack.setName(terminatedName);
+                stack.setTerminated(clock.getCurrentTimeMillis());
+                stackService.save(stack);
+                stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.DELETE_COMPLETED, statusReason);
+                if (stack.getType().equals(StackType.WORKLOAD)) {
+                    grpcUmsClient.notifyResourceDeleted(stack.getResourceCrn(), MDCUtils.getRequestId());
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TerminationService.java:124]: are you sure you want to include an external call in the transaction? I'm against it

Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TerminationService.java]: oh god! :) thanks..

**Review by lacikaaa — APPROVED** [review]

**Review by biharitomi — APPROVED** [review]

**Review by biharitomi — CHANGES_REQUESTED** [review]
https://github.com/hortonworks/cloudbreak/blob/75341a4ccae88235beb652b4e51ba8cc5b131152/core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/terminate/handler/TerminateExternalDatabaseHandler.java#L79
based on the code one more case should be covered what could have caused the stucked stack.
Theoretically the deletion of the external database should not cause the termination of the cluster's stack 😄  Probably a different state was intended to cover those cases

**Comment by sodre90** [comment]
> https://github.com/hortonworks/cloudbreak/blob/75341a4ccae88235beb652b4e51ba8cc5b131152/core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/terminate/handler/TerminateExternalDatabaseHandler.java#L79
> 
> 
> based on the code one more case should be covered what could have caused the stucked stack.
> Theoretically the deletion of the external database should not cause the termination of the cluster's stack  Probably a different state was intended to cover those cases

I will create a separate ticket for this.

**Review by biharitomi — APPROVED** [review]

**Pull request merged**